packages/squiggle-lang/src/rescript/Distributions/SampleSetDist/KdeLibrary.js

```diff
@@ -1,33 +1,75 @@
-const pdfast = require("pdfast");
 const filter = require("lodash/filter");
 const isFinite = require("lodash/isFinite");
 
-const samplesToContinuousPdf = (
-  samples,
-  size,
-  width,
-  min = false,
-  max = false
-) => {
-  let _samples = filter(samples, isFinite);
-  if (isFinite(min)) {
-    _samples = filter(_samples, (r) => r > min);
-  }
-  if (isFinite(max)) {
-    _samples = filter(_samples, (r) => r < max);
-  }
-
-  // The pdf that's created from this function is not a pdf but a pmf. y values
-  // being probability mass and not density.
-  // This is awkward, because our code assumes later that y is a density
-  let pdf = pdfast.create(_samples, { size, width });
-
-  // To convert this to a density, we need to find the step size. This is kept
-  // constant for all y values
-  let stepSize = pdf[1].x - pdf[0].x;
-
-  // We then adjust the y values to density
-  return { xs: pdf.map((r) => r.x), ys: pdf.map((r) => r.y / stepSize) };
+// Convert samples to x-y pairs for a PDF
+// Uses kernel density estimation (KDE) with a triangular kernel
+// samples: Must be sorted!
+// size:    Number of points in output
+// width:   Width of the kernel, as a whole number of x steps
+const samplesToContinuousPdf = (samples, size, width) => {
+  samples = filter(samples, isFinite); // Not sure if this is needed?
+  const len = samples.length;
+  if (len === 0) return { xs: [], ys: [] };
+
+  // To give room for all the kernels, we'll have width steps
+  // on either side as padding.
+  // Subtract from the size - 1 total intervals.
+  const stepsInside = size - 1 - width * 2;
+  if (stepsInside < 1)
+    throw Error("Number of steps isn't enough to contain all values");
```

Comment:
I imagine we'd prefer to return error objects, instead of throwing, more similar to Rescript. @berekuk might have a better idea here.

I'm not a big fan of using "throw" often

Reply:
The caller shouldn't be passing such a small step size or large width, I think. We could also just increase size or decrease width until this works. I need to take a look at what the width recommendation does, I think. It's weird that it gives a width without knowing what actual length it will correspond to.

Reply:
After changing the kernel size argument from a step count to x units, the only requirement is that `outputXYPoints` is at least 3, which is easy to enforce from the outside.